node/src/accountant/mod.rs

```diff
@@ -2730,39 +3130,117 @@ mod tests {
+                .lock()
+                .unwrap();
+        assert!(
+            scan_for_new_payables_notify_later_params.is_empty(),
+            "We did not expect any later scheduling of new payables, but it happened {:?}",
+            scan_for_new_payables_notify_later_params
+        );
+        let scan_for_new_payables_notify_params =
+            scan_for_new_payables_notify_params_arc.lock().unwrap();
+        assert!(
+            scan_for_new_payables_notify_params.is_empty(),
+            "We did not expect any immediate scheduling of new payables, but it happened {:?}",
+            scan_for_new_payables_notify_params
+        );
+        let scan_for_retry_payables_notify_params =
+            scan_for_retry_payables_notify_params_arc.lock().unwrap();
+        assert_eq!(
+            *scan_for_retry_payables_notify_params,
+            vec![ScanForRetryPayables {
+                response_skeleton_opt: None
+            }],
+        );
+    }
+
+    fn assert_receivable_scanner(
+        test_name: &str,
+        earning_wallet: Wallet,
+        receivable_start_scan_params_arc: Arc<
+            Mutex<Vec<(Wallet, SystemTime, Option<ResponseSkeleton>, Logger, String)>>,
+        >,
+        receivable_scan_interval: Duration,
+        scan_for_receivables_notify_later_params_arc: Arc<
+            Mutex<Vec<(ScanForReceivables, Duration)>>,
+        >,
+    ) {
+        let mut receivable_start_scan_params = receivable_start_scan_params_arc.lock().unwrap();
+        let (r_wallet, _r_started_at, r_response_skeleton_opt, r_logger, r_trigger_msg_name_str) =
+            receivable_start_scan_params.remove(0);
+        assert_eq!(r_wallet, earning_wallet);
+        assert_eq!(r_response_skeleton_opt, None);
+        assert!(
+            r_trigger_msg_name_str.contains("Receivable"),
+            "Should contain Receivable but {}",
+            r_trigger_msg_name_str
+        );
+        assert!(
+            receivable_start_scan_params.is_empty(),
+            "Should be already empty but was {:?}",
+            receivable_start_scan_params
+        );
+        assert_using_the_same_logger(&r_logger, test_name, Some("r"));
+        let scan_for_receivables_notify_later_params =
+            scan_for_receivables_notify_later_params_arc.lock().unwrap();
+        assert_eq!(
+            *scan_for_receivables_notify_later_params,
+            vec![(
+                ScanForReceivables {
+                    response_skeleton_opt: None
                 },
                 receivable_scan_interval
             )]
         );
-        // Finally, an assertion to prove course of actions in time. We expect the pending payable
-        // scanner to take place before the payable scanner.
-        // I do believe it's impossible that these two instants would happen at the same time, until
-        // this is run on a real super-giga-computer.
-        assert!(pp_scan_started_at < p_scheduling_now);
```

Comment:
Would you mind sharing your thoughts on whether or not we still need this assertion?